runtime/testy/stateManager.py

```diff
@@ -64,24 +66,29 @@ def addPipe(self, processName, pipe):
 
   def createKey(self, keys):
     currDict = self.state
+    path = []
     for key in keys:
       try:
         if key not in currDict:
-          currDict[key] = {}
-        currDict = currDict[key]
+          currDict[key] = [{}, 0]
+        path.append(currDict[key])
+        currDict = currDict[key][0]
       except TypeError:
         error = StudentAPIKeyError(
           "key '{}' is defined, but does not contain a dictionary.".format(key))
         self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(error)
         return
+    currTime = time.time()
+    for item in path:
```

Comment:
This means we iterate through the path twice. We can accomplish the same results in just a single pass, by updating the timestamp as we iterate through the first time. This will allow you to remove the `path` instance variable you created, as well. 

So, call `currTime = time.time()` before iterating through keys. Before the `currDict = currDict[key][0]` line, add `currDict[key][1] = currTime`. 
Or something like that; I may have an off by one error.